theano/tensor/sort.py

```diff
@@ -206,3 +219,311 @@ def argsort(a, axis=-1, kind='quicksort', order=None):
+
+    def __init__(
+            self,
+            axis=-1,
+            idx_dtype='int64',
+            return_values=True,
+            return_indices=True
+            ):
+        if not isinstance(axis, int):
+            raise TypeError(
+                '"axis" parameter must be integer, got "%s"' % type(axis))
+        if idx_dtype not in theano.tensor.integer_dtypes:
+            raise TypeError(
+                '"idx_dtype" parameter must be an integer dtype, got "%s"' % idx_dtype)
+
+        if not (return_indices or return_values):
+            raise ValueError("Neither return_values nor return_indices is True, this isn't allowd")
+
+        self.axis = axis
+        self.return_values = return_values
+        self.return_indices = return_indices
+        self.idx_dtype = idx_dtype
+
+    def __str__(self):
+        return '%(op)s{axis=%(axis)d}' % dict(
+            op=self.__class__.__name__, axis=self.axis)
+
+    def make_node(self, inp, kth):
+        # numpy always uses int64 as output dtype for arg*() routines
+        # however, we add this option as memory is more precious on gpu
+        inp = theano.tensor.as_tensor_variable(inp)
+        ndim = inp.ndim
+        if ndim == 0:
+            raise ValueError('Cannot take scalar as input')
+        if not -ndim <= self.axis < ndim:
+            raise IndexError(
+                '"axis" parameter out of range,'
+                ' expected integer within [%d, %d]' % (-ndim, ndim - 1))
+
+        kth = theano.tensor.as_tensor_variable(kth)
+        bcast = inp.type.broadcastable
+        outs = []
+        if self.return_values:
+            outs.append(inp.type())
+        if self.return_indices:
+            outs.append(theano.tensor.TensorType(
+                dtype=self.idx_dtype, broadcastable=bcast)())
+        return theano.Apply(self, [inp, kth], outs)
+
+    def perform(self, node, inputs, output_storage):
+        x, k = inputs
+        axis = self.axis
+        if not self.return_indices:
+            pzv = output_storage[0]
+            pzv[0] = _topk_py_impl(self, x, k, axis, None)
+        elif self.return_values:
+            pzv = output_storage[0]
+            pzi = output_storage[1]
+            pzv[0], pzi[0] = _topk_py_impl(
+                self, x, k, axis, node.outputs[1].dtype)
+        else:
+            pzi = output_storage[0]
+            pzi[0] = _topk_py_impl(self, x, k, axis, node.outputs[0].dtype)
+
+    def infer_shape(self, node, inp_shapes):
+        _check_tensor_is_scalar(node.inputs[1])
+        shp = list(inp_shapes[0])
+        shp[self.axis] = np.abs(node.inputs[1])
+        shp = tuple(shp)
+        return [shp for i in [self.return_values, self.return_indices] if i]
+
+    def L_op(self, inputs, outputs, out_grads):
+        x, k = inputs
+        k_grad = grad_undefined(self, 1, k, 'topk: k is not differentiable')
+        if not (self.return_indices, self.return_values):
```

Comment:
this if will always be False. not (non empty tuple) will evaluate like not True, as non-empty tuple always eval to True. You must do:

if not self.return... and not self.return...: